vcd/resource_vcd_subscribed_catalog.go

```diff
@@ -0,0 +1,736 @@
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "A subscribed catalog description is inherited from the publisher catalog and cannot be changed. It is updated on sync",
+			},
+			"storage_profile_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional storage profile ID",
+			},
+			"delete_force": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When destroying use delete_force=True with delete_recursive=True to remove a catalog and any objects it contains, regardless of their state.",
+			},
+			"delete_recursive": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When destroying use delete_recursive=True to remove the catalog and any objects it contains that are in a state that normally allows removal.",
+			},
+			"subscription_url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The URL to subscribe to the external catalog.",
+			},
+			"subscription_password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true,
+				Sensitive: true,
+				// Those unusual password rules are dictated by the API
+				// https://developer.vmware.com/apis/1260/vmware-cloud-director/doc/doc//types/ExternalCatalogSubscriptionParamsType.html
+				Description: "An optional password to access the catalog. " +
+					"Only ASCII characters are allowed in a valid password. " +
+					"Passing in six asterisks '******' indicates to keep current password. " +
+					"Passing in null or empty string indicates to remove password.",
+			},
+			"make_local_copy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Description: "If true, subscription to a catalog creates a local copy of all items. " +
+					"Defaults to false, which does not create a local copy of catalogItems unless sync operation is performed. ",
+			},
+			"metadata": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "Key and value pairs for catalog metadata. Inherited from publishing catalog",
+			},
+			"href": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Catalog HREF",
+			},
+			"created": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Time stamp of when the catalog was created",
+			},
+			"catalog_version": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Catalog version number. Inherited from publishing catalog and updated on sync.",
+			},
+			"owner_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Owner name from the catalog.",
+			},
+			"number_of_vapp_templates": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of vApps templates this catalog contains.",
```

Comment:
Nit, just to match docs

```suggestion
				Description: "Number of vApp templates this catalog contains.",
```